Coding/fw/lib/Lawicel/Lawicel.h

```diff
@@ -0,0 +1,157 @@
+/***************************************************************************************************
+  (c) NewTec GmbH 2020   -   www.newtec.de
+  $URL: https://github.com/NewTec-GmbH/esp32-can-iot $
+***************************************************************************************************/
+/**
+@addtogroup Lawicel
+@{
+@file       Lawicel.h
+
+Driver for Lawicel Protocol
+
+* @}
+***************************************************************************************************/
+#ifndef LAWICEL_H_
+#define LAWICEL_H_
+
+/* INCLUDES ***************************************************************************************/
+#include <Arduino.h>
+#include <string.h>
+#include <CANInterface.h>
+#include <SerialInterface.h>
+#include <NVMInterface.h>
+
+#define MAX_TIMESTAMP 0xEA5F    /**< Maximum value of a Timestamp, representing 1 Minute. */
+#define X_VERSION "V0101"       /**< Hardware and Software Version. */
+#define X_SERIAL_NUMBER "NNT32" /**< Hardware Serial Number. */
+#define CR 13                   /**< CR Character as OK Line Terminator */
+#define BELL 7                  /**< BEL Character as ERROR Line Terminator */
+
+/**
+* Key Definition (Adresses) to be read by NVM Adapter for initialization
+*/
+#define INIT_SERIAL_BAUD "SerialBaud"
+#define INIT_TIMESTAMP "ToggleTime"
+#define INIT_AUTO_START "AutoStart"
+#define INIT_CAN_BAUD "CanBaud"
+#define INIT_FILTER_MODE "FilterMode"
+#define INIT_FILTER_ACN "FilterACn"
+#define INIT_FILTER_AMN "FilterAMn"
+
+/* C-Interface ************************************************************************************/
+extern "C"
+{
+}
+
+/* FORWARD DECLARATIONS ***************************************************************************/
+
+class Lawicel
+{
+public:
+    /* CONSTANTS ******************************************************************************/
+
+    /* TYPES **********************************************************************************/
+
+    /**
+    * Constructor
+    */
+    Lawicel(SerialInterface *_serialInt, CANInterface *_canInt, NVMInterface *_nvmInt) : m_selectedSerial(_serialInt),
+                                                                                         m_selectedCAN(_canInt),
+                                                                                         m_selectedNVM(_nvmInt)
+    {
+    }
+
+    /**
+     *  Default Destructor
+     */
+    ~Lawicel()
+    {
+    }
+
+    uint8_t handler(); /**< Handles the Serial Messages */
```

Comment:
return value ?

handler() is a weak name. Handler for what ? Given the comments, I would call it "processSerialMessages()", or  "executeCycle()"